app/assets/javascripts/users.js

```diff
@@ -0,0 +1,75 @@
+$(document).ready(function() {
+	nameState = false;
+	emailState = false;
+	passwordState = false;
+
+	$("#show_password").change(function() {
+		if ($(this).is(":checked"))
+		{
+			$(".pword").attr("type","text");
+		}
+		else
+		{
+			$(".pword").attr("type","password");
+		}
+	});
+
+	$("#user_first_name, #user_last_name").on("focusout keyup", function() {
+		if ( $(this).val() === "")
+		{
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Field is required");
+			nameState = false;
+		}
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			nameState = true;
+		}
+			changeButton();
+	});
+
+	$("#user_email").on("focusout keyup", function() {
+		var pattern = /^([a-z\d_\.\-])+\@(([a-z\d\-])+\.)+([a-z]{2,4})+$/i;
+		var email = $(this).val();
+		if (!pattern.test(email)) {
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Email is invalid");
+			emailState = false;
+		}
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			emailState = true;
+		}
+		changeButton();
+	});
+
+	$("#user_password").on("focusout keyup", function() {
+		var password = $(this).val();
+		if (password.length < 6){
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Password " +
+				"must be more than 6 characters");
+			passwordState = false;
+		}
+
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			passwordState = true;
+		}
+		changeButton();
+	});
+
+	function changeButton () {
+		if (passwordState && emailState && nameState)
```

Comment:
'passwordState' is not defined.<br>'emailState' is not defined.<br>'nameState' is not defined.